compiler/ballerina-lang/src/main/java/io/ballerina/projects/bala/BalaProject.java

```diff
@@ -126,4 +126,9 @@ private boolean isFilePathInProject(Path filepath) {
         }
         return true;
     }
+
+    @Override
+    public Path targetDir() {
+        throw new UnsupportedOperationException("Target Dir is not supported for BalaProject");
```

Comment:
I have second thoughts on this. Should we rather return `null` and add it in javadocs for the method? @sameerajayasoma @hevayo wdyt?

Reply:
We'll go ahead with throwing the exception.